src/main/java/org/hawkular/client/android/activity/DrawerActivity.java

```diff
@@ -66,8 +78,67 @@ private void setUpNavigation() {
         navigation.setNavigationItemSelectedListener(this);
     }
 
+    private void setUpBackendClient() {
+        String backendHost = Preferences.of(this).host().get();
+        int backendPort = Preferences.of(this).port().get();
+
+        if (backendHost.isEmpty() || backendPort == Preferences.Defaults.BACKEND_PORT) {
+            startAuthorizationActivity();
+            return;
+        }
+
+        BackendClient.of(this).configureBackend(backendHost, backendPort);
+
+        BackendClient.of(this).authorize(this, this);
+    }
+
+    @Override
+    public void onSuccess(String authorization) {
+        setUpDefaults();
+    }
+
+    private void setUpDefaults() {
+        showNavigation(R.id.menu_resources);
+    }
+
+    private void showNavigation(@IdRes int navigationId) {
+        onNavigationItemSelected(navigation.getMenu().findItem(navigationId));
+    }
+
+    @Override
+    public void onFailure(Exception e) {
+        startAuthorizationActivity();
+    }
+
+    private void startAuthorizationActivity() {
+        Intent intent = Intents.Builder.of(this).buildAuthorizationIntent();
+        startActivityForResult(intent, Intents.Requests.AUTHORIZATION);
+    }
+
+    @Override
+    protected void onActivityResult(int request, int result, Intent intent) {
+        super.onActivityResult(request, result, intent);
+
+        if ((request == Intents.Requests.AUTHORIZATION) && (result != RESULT_OK)) {
+            finish();
```

Comment:
Nope, wrong credentials will show an error at the `AuthorizationActivity`. Only if user will press the back button at the `AuthorizationActivity`, i. e. deny to authorize the application will close, mostly because we have nothing to show to him.